src/main/java/meteordevelopment/meteorclient/systems/modules/render/Tracers.java

```diff
@@ -192,6 +262,141 @@ else if (entity instanceof PlayerEntity) {
+                color = PlayerUtils.getPlayerColor(((PlayerEntity) entity), playersColor.get());
+            }
+            else {
+                color = switch (entity.getType().getSpawnGroup()) {
+                    case CREATURE -> animalsColor.get();
+                    case WATER_AMBIENT, WATER_CREATURE, UNDERGROUND_WATER_CREATURE, AXOLOTLS -> waterAnimalsColor.get();
+                    case MONSTER -> monstersColor.get();
+                    case AMBIENT -> ambientColor.get();
+                    default -> miscColor.get();
+                };
+            }
+
+            Color finalColor = new Color(color.r, color.g, color.b, color.a);
+            if (blinkOffscreen.get())
+                finalColor.a *= getAlpha();
+
+            final Vec2f SCREEN_CENTER = new Vec2f(mc.getWindow().getFramebufferWidth() / 2.f, mc.getWindow().getFramebufferHeight() / 2.f);
+
+            Vector3d projection = new Vector3d(entity.prevX, entity.prevY, entity.prevZ);
+            boolean projSucceeded = NametagUtils.to2D(projection, 1, false, false);
+
+            if (projSucceeded && projection.x > 0.f && projection.x < mc.getWindow().getFramebufferWidth() && projection.y > 0.f && projection.y < mc.getWindow().getFramebufferHeight())
+                continue;
+
+            projection = new Vector3d(entity.prevX, entity.prevY, entity.prevZ);
+            NametagUtils.to2D(projection, 1, false, true);
+
+            Vector2f angle = VectorAngles(new Vector3d(SCREEN_CENTER.x - projection.x, SCREEN_CENTER.y - projection.y, 0));
+            angle.y += 180;
+
+            MatrixStack matrices = RenderSystem.getModelViewStack();
+            matrices.push();
+            matrices.translate(0, 0, 0);
+            matrices.scale(1.f, 1.f, 1);
+
+            final float ANGLE_YAW_RAD = (float) Math.toRadians(angle.y);
+
+            final Vector2f NEW_POINT = new Vector2f(SCREEN_CENTER.x + distanceOffscreen.get() * (float) Math.cos(ANGLE_YAW_RAD),
+                SCREEN_CENTER.y + distanceOffscreen.get() * (float) Math.sin(ANGLE_YAW_RAD));
+
+            Vector2f TRIANGLE_POINTS[] = {
+                new Vector2f(NEW_POINT.x - sizeOffscreen.get(), NEW_POINT.y - sizeOffscreen.get()),
+                new Vector2f(NEW_POINT.x + sizeOffscreen.get() * 0.73205f, NEW_POINT.y),
+                new Vector2f(NEW_POINT.x - sizeOffscreen.get(), NEW_POINT.y + sizeOffscreen.get())
+            };
+
+            // rotate
+            rotateTriangle(TRIANGLE_POINTS, angle.y);
+
+            Renderer2D.COLOR.begin();
+            Renderer2D.COLOR.triangle(TRIANGLE_POINTS[0].x, TRIANGLE_POINTS[0].y, TRIANGLE_POINTS[1].x, TRIANGLE_POINTS[1].y, TRIANGLE_POINTS[2].x,
+                TRIANGLE_POINTS[2].y, finalColor);
+            Renderer2D.COLOR.render(null);
+
+            count++;
+
+            RenderSystem.getModelViewStack().pop();
+        }
+    }
+
+    private Vector2f addVec2f(Vector2f a, Vector2f b) {
```

Comment:
`Vector2f` already contains these methods, either `a.add(b, new Vector2f())` to create a new one or if you omit the second parameter then it will store the result inside `a` for better performance